### PR TITLE
Updating Key Vault commands to use REST API v2015-06-01

### PR DESCRIPTION
**Pull request by msfcolombo**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@msfcolombo__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (fcolombo). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/keyvault/kv-legacy.js [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+*/
+KeyVaultKeys.prototype.backup = function(keyIdentifier, callback) {
+    return this.keys.backup(keyIdentifier, keyResponseDeserializer(callback));
+};
+
+/**
+* Creates a new, named, key in the specified vault.
+* @param {string} vault
+* @param {string} keyName
+* @param {object} keyOpRequest
```
Comment (amarzavery) [inline lib/commands/arm/keyvault/kv-legacy.js]: should there be documentation on the keys of this object?

Comment (msfcolombo) [inline lib/commands/arm/keyvault/kv-legacy.js]: This file was copied from azure-keyvault 0.9.3. The jsdoc here is actually useless.

**Comment by msfcolombo** [comment]
I'm not sure why the build passed on 0.10 and 0.12, and failed on 0.8. Doesn't seem related to my changes. How can we retry the build?

**Comment by yugangw-msft** [comment]
This is a known test flakiness introduced by the recent webapp cmdlet. I am looking at it. So don't worry.

**Comment by yugangw-msft** [comment]
please pull from dev and update the changelog.txt

**Comment by msfcolombo** [comment]
thanks, pulled dev and changelog.txt updated.

File: lib/commands/arm/keyvault/kv-legacy.js [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+};
+
+function serializeSecretRequest(request) {
+    return {
+        rawJsonRequest: JSON.stringify(request)
+    };
+}
+
+function secretResponseDeserializer(callback) {
+    return function(err, response) {
```
Comment (yugangw-msft) [inline lib/commands/arm/keyvault/kv-legacy.js]: please fix the indent

File: lib/commands/arm/keyvault/keyvault.key._js [hunk 3]
```diff
@@ -35,7 +36,7 @@ var KEY_DEST_TYPE_MAP = {
 };
 var KEY_DESTS = Object.keys(KEY_DEST_TYPE_MAP);
 
-var KEY_OPS = ['encrypt', 'decrypt', 'wrap', 'unwrap', 'sign', 'verify', 'backup', 'restore'];
+var KEY_OPS = ['encrypt', 'decrypt', 'sign', 'verify', 'wrapKey', 'unwrapKey'];
```
Comment (yugangw-msft) [inline lib/commands/arm/keyvault/keyvault.key._js]: could you run "bin/azure --gen arm" to see whether this updates the command "metadata"? If yes, you will see a "plugins.arm" checked out, you just include that file in this PR

**Comment by yugangw-msft** [comment]
lgtm. Please address 2 extra nits and we can merge

File: lib/commands/arm/keyvault/keyvault.key._js (hunk 3 above)
Comment (msfcolombo) [inline lib/commands/arm/keyvault/keyvault.key._js]: Verified that it doesn't change the plugins.arm file.

**Comment by msfcolombo** [comment]
done again, nits fixed

**Comment by yugangw-msft** [comment]
I can't merge this Pr due to conflict, please pull and resolve them

**Pull request merged**